crypto/params.c

```diff
@@ -608,13 +608,15 @@ OSSL_PARAM OSSL_PARAM_construct_size_t(const char *key, size_t *buf)
 int OSSL_PARAM_get_BN(const OSSL_PARAM *p, BIGNUM **val)
 {
     BIGNUM *b;
+    size_t len;
 
     if (val == NULL
         || p == NULL
         || p->data_type != OSSL_PARAM_UNSIGNED_INTEGER)
         return 0;
 
-    b = BN_native2bn(p->data, (int)p->data_size, *val);
+    len = p->return_size > 0 ? p->return_size : p->data_size;
```

Comment:
You do remember that this is dicy, right? The "other" end may not have initialized *return_size* at all...

Reply:
It would be better to have a separate function for the OSSL_PARAM array owner. I haven't been able to figure out a suitable name, though, and mostly figured it would just be an alias for `BN_native2bn` in the end...

Reply:
Please suggest an alternative way for a OSSL_PARAM with a 50 byte buffer to accept a 20 byte padded BN?

I could have `OSSL_PARAM_set_BN_pad()` zero fill the OSSL_PARAM buffer but that's doesn't seem like constant time which is the entire point of wanting this new function.

Reply:
I have *no* idea what you're commenting on here.  This is `OSSL_PARAM_get_bn()`, which is, or at least was designed for the "receiving end" (i.e. the type of function that receives an `OSSL_PARAM` pointer as an argument) of the `OSSL_PARAM` array.  As such, it should only rely on *data_size*.

I assume you're again trying to redesign it for double purposes, i.e. to become something for the "sending end" (i.e. the `OSSL_PARAM` owner) to use.  I'm only pointing out that there's no guarantee that the "sending end" has zeroed *return_size*, and the "receiving end" shouldn't have to depend on that.